skynet/agent.py

```diff
@@ -41,3 +49,75 @@ async def create_tcpserver(port):
         await tcpserver.serve_forever()
     except KeyboardInterrupt:
         tcpserver.close()
+
+
+# Endpoints for the autoscaler to query the current state of the system
+class StateResponse(BaseModel):
+    request_status: str
+    current_state: str
+    desired_state: str
+
+
+class CurrentStateResponse(BaseModel):
+    current_state: str
+    active_connections: int
+    stress_level: float
+
+
+async def get_current_state():
+    try:
+        with open(STATE_FILE, 'r') as f:
```

Comment:
Rather than opening the file twice in case it's not there, you can open it for reading _and_ writing, and return the content if you find any, or write it if missing.

Not sure how often that operation happens though...